### PR TITLE
Add "Sponsored by NumFOCUS" badge

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

I found the "Powered by NumFOCUS" badge when looking at other projects and will add it.

 
 relate to #3999

### Details

- None

### CODE REVIEW

**Review by banesullivan — CHANGES_REQUESTED** [review]
Considering that PyVista is only an "Affiliated Project" at this time (as opposed to a "Sponsored Project"), I think we should choose different phrasing than "Powered by". We should look to other affiliated projects for guidance on the phrases and/or potential badges to re-use

**Comment by akaszynski** [comment]
I looked at https://numfocus.org/sponsored-projects/affiliated-projects and after checking ~5 I didn't see any with this badge. Agree with @banesullivan, let's put this on hold.

I'd be happy with a "Affiliated with Numfocus" badge.

**Review by akaszynski — APPROVED** [review]
LGTM. Thanks @tkoyama010!

File: README.rst [hunk 1]
```diff
@@ -54,24 +54,29 @@ PyVista
...
-| Citation             | |joss|    |  |zenodo|  |
-+----------------------+-----------+------------+
-| License              | |MIT|     |            |
-+----------------------+-----------+------------+
-| Community            | |slack|   |  |discuss| |
-+----------------------+-----------+------------+
-| Formatter            | |black|   |  |isort|   |
-+----------------------+-----------+------------+
+.. |Sponsored by NumFOCUS| image:: https://img.shields.io/badge/sponsored%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A
+   :target: https://numfocus.org
```
Comment (banesullivan) [inline README.rst:58]: I'm curious why this was merged with the "Sponsored by" language still present? This should be changed to:

```suggestion
.. | Affiliated with NumFOCUS| image:: https://img.shields.io/badge/affiliated-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A
   :target: https://numfocus.org
```

**Pull request merged**